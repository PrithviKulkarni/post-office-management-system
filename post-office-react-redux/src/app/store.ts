import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
