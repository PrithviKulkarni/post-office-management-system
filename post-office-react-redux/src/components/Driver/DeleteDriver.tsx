import { SyntheticEvent } from "react";
import { useParams, useNavigate } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "../../app/hooks";
import { deleteADriver, getDriverById } from "../../features/DriverSlice";
import { RootState } from "../../app/store";
import { Driver } from "../interfaces/driverinterface";
import Button from "react-bootstrap/Button";
import Card from 'react-bootstrap/Card';
import CardGroup from 'react-bootstrap/CardGroup';

const DeleteDriver = () => {
  const params = useParams();
  const dispatch = useAppDispatch();
  const navigate = useNavigate();
  const driver: Driver | undefined = useAppSelector((state: RootState) =>
    getDriverById(state, params.driverId!)
  );


  const submitHandler = (event: SyntheticEvent) => {
      dispatch(deleteADriver(driver!));
      navigate("/drivers");
  };

  return (
  <div className="delete-driver" data-testid="delete-driver-test">
      <CardGroup>
      <Card key={driver?.id}>
        <Card.Body>
          <Card.Title>{driver?.first_name} &nbsp; {driver?.last_name}</Card.Title>
          <Card.Text>
            {driver?.email}
          </Card.Text>
        </Card.Body>
        <Card.Footer>
        <Button onClick={submitHandler}>Confirm Delete</Button>
        </Card.Footer>
      </Card>
      </CardGroup>
  </div>
  );
}

export default DeleteDriver
