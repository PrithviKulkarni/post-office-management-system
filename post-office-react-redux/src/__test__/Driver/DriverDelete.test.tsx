import { render, cleanup } from '@testing-library/react';
import { screen } from '@testing-library/react';
import { Provider } from "react-redux";
import { store } from '../../app/store';
import { BrowserRouter } from "react-router-dom";
import DeleteDriver from '../../components/Driver/DeleteDriver'

afterEach(cleanup);

// DELETE DRIVERS COMPONENT

it('Renders Delete Drivers Page Correctly', () => {
     const { asFragment } = render(
          <Provider store={store}>
               <BrowserRouter>
                    <DeleteDriver />
               </BrowserRouter>
          </Provider>
     );
     expect(asFragment()).toMatchSnapshot();
});

it('Should Have Div ID delete-driver', () => {
     render(
          <Provider store={store}>
               <BrowserRouter>
                    <DeleteDriver />
               </BrowserRouter>
          </Provider>
     );
     expect(screen.getByTestId("delete-driver-test")).toHaveClass("delete-driver");

});


