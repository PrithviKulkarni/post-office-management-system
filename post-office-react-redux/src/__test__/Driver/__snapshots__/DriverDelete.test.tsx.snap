// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Delete Drivers Page Correctly 1`] = `
<DocumentFragment>
  <div
    class="delete-driver"
    data-testid="delete-driver-test"
  >
    <div
      class="card-group"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
             Â  
          </div>
          <p
            class="card-text"
          />
        </div>
        <div
          class="card-footer"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            Confirm Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
